internal/mode/static/nginx/config/servers_test.go

```diff
@@ -3052,33 +2984,199 @@ func TestGenerateProxySetHeaders(t *testing.T) {
 					Name:  "my-header",
 					Value: "",
 				},
+			}, grpcBaseHeaders...),
+			baseHeaders: grpcBaseHeaders,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.msg, func(t *testing.T) {
+			t.Parallel()
+			g := NewWithT(t)
+
+			headers := generateProxySetHeaders(tc.filters, tc.baseHeaders)
+			g.Expect(headers).To(Equal(tc.expectedHeaders))
+		})
+	}
+}
+
+func TestCreateBaseProxySetHeaders(t *testing.T) {
+	t.Parallel()
+
+	expBaseHeaders := []http.Header{
+		{
+			Name:  "Host",
+			Value: "$gw_api_compliant_host",
+		},
+		{
+			Name:  "X-Forwarded-For",
+			Value: "$proxy_add_x_forwarded_for",
+		},
+		{
+			Name:  "X-Real-IP",
+			Value: "$remote_addr",
+		},
+		{
+			Name:  "X-Forwarded-Proto",
+			Value: "$scheme",
+		},
+		{
+			Name:  "X-Forwarded-Host",
+			Value: "$host",
+		},
+		{
+			Name:  "X-Forwarded-Port",
+			Value: "$server_port",
+		},
+	}
+
+	tests := []struct {
+		msg               string
+		additionalHeaders []http.Header
+		expBaseHeaders    []http.Header
+	}{
+		{
+			msg:               "no additional headers",
+			additionalHeaders: []http.Header{},
+			expBaseHeaders:    expBaseHeaders,
+		},
+		{
+			msg: "single additional headers",
+			additionalHeaders: []http.Header{
+				grpcAuthorityHeader,
+			},
+			expBaseHeaders: append(expBaseHeaders, grpcAuthorityHeader),
+		},
+		{
+			msg: "multiple additional headers",
+			additionalHeaders: []http.Header{
+				httpConnectionHeader,
+				httpUpgradeHeader,
+			},
+			expBaseHeaders: append(expBaseHeaders, httpConnectionHeader, httpUpgradeHeader),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.msg, func(t *testing.T) {
+			t.Parallel()
+			g := NewWithT(t)
+
+			result := createBaseProxySetHeaders(test.additionalHeaders...)
+			g.Expect(result).To(Equal(test.expBaseHeaders))
+		})
+	}
+}
+
+func TestGetConnectionHeader(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		msg                 string
+		upstreams           []http.Upstream
```

Comment:
Since you are testing the keepAliveChecker functionality in the upstreams test, you don't need to test it again here. You can just mock the keepAliveChecker by passing a function that matches the keepAliveChecker signature. That will simplify these tests.

Reply:
I thought about mocking it since I know I already tested the keepAliveChecker functionality in the upstreams test. 

This is a little confusing but I think this is correct and warrants the non-mocking of the function. The other test simply checks for a single upstream, does that upstream have keepAlive enabled. This test combines that with if any of the upstream -> backend names have keepalive, does it return the right header. 

So I guess I could mock the keepAliveChecker, but I would have an array of keepAliveCheckers that return a mix of false / true ya know? Have I misunderstood something or does my point come across correctly?

Reply:
ok, ill keep it as is unless there are other thoughts, thanks for suggesting